File: github_javapop/app/src/main/java/com/lfernando/githubjavapop/network/GithubApi.java
package com.lfernando.githubjavapop.network;

import com.lfernando.githubjavapop.model.PullRequest;
import com.lfernando.githubjavapop.model.Repo;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface GithubApi {
    @GET("search/repositories?q=language:Java&sort=stars")
    Call<RepositoryReponse> listRepos(@Query("page") String pageNumber);

    @GET("repos/{owner}/{repo}/pulls")
    Call<List<PullRequest>> listPrs(@Path("owner") String owner, @Path("repo") String repository);
}

File: GithubSearch/app/src/main/java/br/com/bernardino/githubsearch/api/GithubApi.kt
package br.com.bernardino.githubsearch.api

import br.com.bernardino.githubsearch.model.PullRequest
import br.com.bernardino.githubsearch.model.Repository
import br.com.bernardino.githubsearch.model.RepositoryBody
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GithubApi {

    @GET("search/repositories")
    fun getRepositories(@Query("q") language: String, @Query("sort") sort: String, @Query("page") page: Int) : Call<RepositoryBody>

    @GET ("repos/{creator}/{repository}/pulls")
    fun getPullRequests (@Path("creator") creator: String?,
                         @Path("repository") repository : String?) : Call<List<PullRequest>>
}